apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -144,10 +130,53 @@ export const FirmwareUpdate = ({
     [device.modelId],
   );
 
+  const [fullUpdateComplete, setFullUpdateComplete] = useState(false);
+
+  const { updateActionState, updateStep, retryUpdate, staxLoadImageState } =
+    useUpdateFirmwareAndRestoreSettings({
+      updateFirmwareAction,
+      device,
+    });
+
+  useEffect(() => {
+    if (updateStep === "completed") {
+      const completeTimeout = setTimeout(
+        () => setFullUpdateComplete(true),
+        3000,
+      );
+
+      return () => clearTimeout(completeTimeout);
+    }
+
+    return undefined;
+  });
+
+  const restoreSteps = useMemo(
+    () => [
+      {
+        status: {
+          start: ItemStatus.inactive,
+          imageBackup: ItemStatus.inactive,
+          firmwareUpdate: ItemStatus.inactive,
+          languageRestore: ItemStatus.inactive,
+          imageRestore: ItemStatus.active,
+          appsRestore: ItemStatus.completed,
+          completed: ItemStatus.completed,
+        }[updateStep],
+        progress: staxLoadImageState.progress,
+        title: t(
+          "FirmwareUpdate.steps.restoreSettings.restoreLockScreenPicture",
```

Comment:
I've always hated that the i18nKey path for the firmware update is capitalized for no reason, +1 for consistency (not you, in general)